biz.aQute.gogo.commands.provider/src/main/java/biz/aQute/gogo/commands/provider/ExportPackageBundleTracker.java

```diff
@@ -0,0 +1,113 @@
+package biz.aQute.gogo.commands.provider;
+
+import java.io.Closeable;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.Constants;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.util.tracker.BundleTracker;
+
+public class ExportPackageBundleTracker extends BundleTracker<Bundle> implements Closeable {
```

Comment:
I do not think this is worth the effort as we discussed. To do this properly you need to do a lot of work and you might still be wrong due to class loader constraints. Just loading a class from the package, protected by a try/catch is the only reliable way to find out you can import something. And it is a lot simpler.

Reply:
I changed it. Now the ExportPackageBundleTracker does no check than the the registration protected by the try/catch. But is tryes it again an new installed bundles.

I am still not happy with the exceptions. but do not see an other way.